include/CppUTest/CppUTestConfig.h

```diff
@@ -134,32 +134,21 @@
  * To check whether it is on or off, we create a CppUTest define here.
 */
 #if defined(__has_feature)
-#if __has_feature(address_sanitizer)
-#define CPPUTEST_SANITIZE_ADDRESS 1
-#endif
-#endif
-
-#ifdef __SANITIZE_ADDRESS__
-#define CPPUTEST_SANITIZE_ADDRESS 1
+  #if __has_feature(address_sanitizer)
+    #define CPPUTEST_SANITIZE_ADDRESS 1
+  #endif
+#elif defined(__SANITIZE_ADDRESS__)
+  #define CPPUTEST_SANITIZE_ADDRESS 1
 #endif
 
 #ifndef CPPUTEST_SANITIZE_ADDRESS
-#define CPPUTEST_SANITIZE_ADDRESS 0
+  #define CPPUTEST_SANITIZE_ADDRESS 0
 #endif
 
 #if CPPUTEST_SANITIZE_ADDRESS
-#define CPPUTEST_SANITIZE_ADDRESS 1
-#define CPPUTEST_DO_NOT_SANITIZE_ADDRESS __attribute__((no_sanitize_address))
-  #if defined(__linux__) && defined(__clang__)
-    #if CPPUTEST_USE_MEM_LEAK_DETECTION
-    #warning Compiling with Address Sanitizer with clang on linux will cause duplicate symbols for operator new. Turning off memory leak detection. Compile with -DCPPUTEST_MEM_LEAK_DETECTION_DISABLED to get rid of this warning.
```

Comment:
In the past with clang, when you turned on address sanitizer then the linker would give duplicate symbol errors on operator new. That is why this was added. If this is not so anymore, then I guess that was fixed in the address sanitizer... however then it would probably be better to have this piece of code behind a version check of clang than to remove it. If we remove it and people use that compiler version then they will get the same errors again.

Reply:
I've run into similar errors working with the memory sanitizer. My approach has been to disable use of the C++ standard library since it isn't instrumented and produces many false positives. I wonder it we should take a similar approach here.

Reply:
I've disabled the the C++ standard library for asan tests and restore the warning with an additional condition of its use.